src/ol/renderer/canvas/canvasvectorlayerrenderer.js

```diff
@@ -194,6 +227,14 @@ ol.renderer.canvas.VectorLayer.prototype.prepareFrame =
 
   var extent = ol.extent.buffer(frameStateExtent,
       vectorLayerRenderBuffer * resolution);
+  var projectionExtent = viewState.projection.getExtent();
+
+  if (vectorSource.getWrapX() && viewState.projection.isGlobal() &&
```

Comment:
What does "global projection" mean in OpenLayers? Why do we have this notion? Is it just for wrapX?

Looking at the code it looks like that it's just for wrapX. So I wonder why we need it. I wonder why we do only rely on the wrapX option to determine if the layer/source should be wrapped.

Maybe I'm missing something...